src/Microsoft.Diagnostics.Runtime/datatarget.cs

```diff
@@ -2045,16 +2058,40 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, uint conte
     internal unsafe class LiveDataReader : IDataReader
     {
         #region Variables
+        private int _originalPid;
+        private IntPtr _snapshotHandle;
+        private IntPtr _vaCloneHandle;
         private IntPtr _process;
         private int _pid;
         #endregion
 
         private const int PROCESS_VM_READ = 0x10;
         private const int PROCESS_QUERY_INFORMATION = 0x0400;
-        public LiveDataReader(int pid)
+        public LiveDataReader(int pid, bool createSnapshot = false)
```

Comment:
Can you make this a non-default parameter?  I'd rather it be clear there's a choice here at the callsite.